io/s3.go

```diff
@@ -45,43 +44,42 @@ const (
 )
 
 func createS3FileIO(parsed *url.URL, props map[string]string) (IO, error) {
-	opts := []func(*config.LoadOptions) error{}
+	cfgOpts := []func(*config.LoadOptions) error{}
+	opts := []func(*s3.Options){}
+
 	endpoint, ok := props[S3EndpointURL]
 	if !ok {
 		endpoint = os.Getenv("AWS_S3_ENDPOINT")
 	}
 
 	if endpoint != "" {
-		opts = append(opts, config.WithEndpointResolverWithOptions(aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
-			if service != s3.ServiceID {
-				// fallback to default resolution for the service
-				return aws.Endpoint{}, &aws.EndpointNotFoundError{}
-			}
-
-			return aws.Endpoint{
-				URL:               endpoint,
-				SigningRegion:     region,
-				HostnameImmutable: true,
-			}, nil
-		})))
+		opts = append(opts, func(o *s3.Options) {
```

Comment:
for the future I think it would be great to just extract such things out into a separate (small) PR. That way we can get PRs reviewed faster as otherwise it's quite difficult to find long periods of time to review a huge chunk of new code that is mixed with other changes

Reply:
Sorry about this, I originally had intended for this to be done as a separate change but the CI started failing without this / I wasn't able to get the CI testing of the new changes without this so I ended up adding it to this.